cosmos/scheduler/cmd/scheduler.go

```diff
@@ -32,39 +35,57 @@ type Scheduler struct {
 	IngestionSchedule string
 }
 
+// BucketTransaction is used as the decoded element returned by MongoDB
+type BucketTransaction struct {
+	LocationType  string
+	OperationType string `bson:"operationType"`
+	FullDocument  struct {
+		Value struct {
+			Name               string `bson:"name"`
+			OwnerDisplayName   string `bson:"ownerDisplayName"`
+			Deleted            bool   `bson:"deleted"`
+			LocationConstraint string `bson:"locationConstraint"`
+		} `bson:"value"`
+	} `bson:"fullDocument"`
+}
+
 // Run starts the Cosmos Scheduler
 func (s *Scheduler) Run() {
 	log.Println("starting cosmos scheduler")
-	quit := make(chan bool)
-	overlayUpdates, err := s.watchOverlayUpdates()
-	if err != nil {
-		log.Println(err)
-		return
-	}
 	go s.healthCheckServer()
-	err = s.configureIngestionSecret(true)
+
+	// safely create or update secret on startup
+	err := s.configureIngestionSecret(true)
 	if err != nil{
-		log.Fatal("error configuring secret:", err)
+		log.Fatal("run failed configuring secret: ", err)
 	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	overlayUpdates := s.watchOverlayUpdates(ctx)
 	for {
-		go s.watchBucketUpdates(quit)
-		<-overlayUpdates
+		ctxNfs, cancelNfs := context.WithCancel(ctx)
+		go s.watchBucketUpdates(ctxNfs, "location-nfs-mount-v1")
+		// blocks until an overlay update is recieved on the overlayUpdates channel
+		<- overlayUpdates
+		cancelNfs()
 		log.Println("received an overlay update")
-		quit <- true
```

Comment:
TODO: the loop is still essentially watch, update, quit. Will still be reworking this.